### ISSUE TITLE
Removing outdated records from the 'AggregatedCounter' table: Query processor could not produce a query plan because of the hints defined in this query.

### ISSUE THREAD

**Issue by ManuelHaas** [HangfireIO/Hangfire#2257]
Hi,

after updating from Hangfire 1.7.34 to 1.8.3 I get lots of the following errors in the log:
```
[09:43:00 DBG] Removing outdated records from the 'AggregatedCounter' table...
[09:43:00 ERR] Execution ExpirationManager is in the Failed state now due to an exception, execution will be retried no more than in 00:00:25
Microsoft.Data.SqlClient.SqlException (0x80131904): Query processor could not produce a query plan because of the hints defined in this query. Resubmit the query without specifying any hints and without using SET FORCEPLAN.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String methodName)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Hangfire.SqlServer.ExpirationManager.ExecuteNonQuery(DbConnection connection, String commandText, Int32 numberOfRecordsInSinglePass, CancellationToken cancellationToken, Action`1 additionalActions)
   at Hangfire.SqlServer.ExpirationManager.<>c__DisplayClass11_0.<CleanupTable>b__0(DbConnection connection)
   at Hangfire.SqlServer.ExpirationManager.<>c__DisplayClass12_0.<UseConnectionDistributedLock>b__0(DbConnection connection)
   at Hangfire.SqlServer.SqlServerStorage.<>c__DisplayClass37_0.<UseConnection>b__0(DbConnection connection)
   at Hangfire.SqlServer.SqlServerStorage.UseConnection[T](DbConnection dedicatedConnection, Func`2 func)
   at Hangfire.SqlServer.SqlServerStorage.UseConnection(DbConnection dedicatedConnection, Action`1 action)
   at Hangfire.SqlServer.ExpirationManager.UseConnectionDistributedLock(SqlServerStorage storage, Action`1 action)
   at Hangfire.SqlServer.ExpirationManager.CleanupTable(String query, String table, Int32 numberOfRecordsInSinglePass, CancellationToken cancellationToken, Action`1 additionalActions)
   at Hangfire.SqlServer.ExpirationManager.Execute(CancellationToken cancellationToken)
   at Hangfire.Server.ServerProcessDispatcherBuilder.ExecuteComponent(Guid executionId, Object state)
   at Hangfire.Processing.BackgroundExecution.Run(Action`2 callback, Object state)
ClientConnectionId:01d4da92-806d-49c5-863a-dd5eabe35174
Error Number:8622,State:1,Class:16
[09:43:00 DBG] Execution loop ExpirationManager:45d2433d caught an exception and will be retried in 00:00:25
```

- The AggregatedCounter table has 163 rows.
- A UX_HangFire_CounterAggregated_Key (Unique, Non-Clustered) index exists on the table.
- SQL Server 2019 (15.0.4280.7) is used
- I have no changes made in Hangfire tables

This is a similar error to #1088 but not the same.

Has anyone an Idea what can be the problem?

**Comment by odinserj** [comment]
Hi @ManuelHaas, thanks for reporting this! I already committed a fix for the issue, it will be released soon in 1.8.5. Alternatively, you can use the CI feed `https://ci.appveyor.com/nuget/hangfire` with the package version `1.8.4-build-03829`.

**Comment by ManuelHaas** [comment]
Thanks for the prompt reply and the fix!
